app/screens/image_preview/previewer.js

```diff
@@ -0,0 +1,277 @@
+            imageHeight: new Animated.Value(props.imageHeight),
+            imageWidth: new Animated.Value(props.imageWidth),
+            opacity: new Animated.Value(0),
+            requesting: true,
+            retry: 0,
+            zooming: false
+        };
+    }
+
+    componentWillMount() {
+        this.panResponder = PanResponder.create({
+            onStartShouldSetPanResponderCapture: (evt, gestureState) => {
+                const {numberActiveTouches} = gestureState;
+                if (numberActiveTouches === 1 && !this.state.isZooming) {
+                    return true;
+                } else if (numberActiveTouches === 1) {
+                    this.handleResponderRelease(evt);
+                }
+
+                return false;
+            },
+            onPanResponderGrant: () => {
+                return;
+            },
+            onPanResponderTerminate: () => {
+                return;
+            },
+            onShouldBlockNativeResponder: () => false
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.shrink && !nextProps.shrink) {
+            this.setShrink();
+        } else if (!this.props.shrink && nextProps.shrink) {
+            this.setShrink(true);
+        }
+    }
+
+    setShrink = (shrink = false) => {
+        const {gutter, imageHeight, imageWidth} = this.props;
+
+        let height = imageHeight;
+        let width = imageWidth;
+        const duration = 150;
+        if (shrink) {
+            height = height - gutter;
+            width = width - gutter;
+        }
+
+        const animations = [
+            Animated.timing(this.state.imageWidth, {
+                toValue: width,
+                duration
+            })
+        ];
+
+        if (Platform.OS === 'android') {
+            animations.push(
+                Animated.timing(this.state.imageHeight, {
+                    toValue: height,
+                    duration
+                })
+            );
+        }
+
+        Animated.parallel(animations).start();
+    }
+
+    handleResponderRelease = (evt) => {
+        clearTimeout(this.singleTap);
+        let cancelSingleTap = false;
+        if (this.lastTap && Date.now() - this.lastTap < DOUBLE_CLICK_THRESHOLD) {
+            cancelSingleTap = true;
+        } else {
+            this.lastTap = Date.now();
+        }
+
+        if (cancelSingleTap) {
+            const {nativeEvent} = evt;
+            const x = nativeEvent.locationX;
+            const y = nativeEvent.locationY;
+
+            cancelSingleTap = false;
+            this.lastTap = null;
+
+            this.props.onImageDoubleTap(x, y);
+        } else if (!this.state.isZooming) {
+            this.singleTap = setTimeout(() => {
+                this.props.onImageTap();
+            }, DOUBLE_CLICK_THRESHOLD);
+        }
+    }
+
+    handleLoadError = () => {
+        if (this.state.retry < 4) {
+            setTimeout(() => {
+                this.setState({
```

Comment:
The image component will try to load the image and if it fails it won't actually run through the loading functions again, it will just grab whatever is cached based on the URL. So by adding a timestamp in handleLoadError it changes the source.uri so that the Image component will actually run through the loading functions again because the uri is different when the image is re-rendered.